ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -29,52 +32,63 @@
 import org.opensearch.ml.common.input.MLInput;
 import org.opensearch.ml.common.output.model.ModelTensorOutput;
 import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.common.transport.MLTaskResponse;
 import org.opensearch.script.ScriptService;
 
 public interface RemoteConnectorExecutor {
 
-    default ModelTensorOutput executePredict(MLInput mlInput) {
-        List<ModelTensors> tensorOutputs = new ArrayList<>();
-
+    default void executePredict(MLInput mlInput, ActionListener<MLTaskResponse> actionListener) {
+        ActionListener<List<ModelTensors>> tensorActionListener = ActionListener.wrap(r -> {
+            actionListener.onResponse(new MLTaskResponse(new ModelTensorOutput(r)));
+        }, actionListener::onFailure);
+        Map<Integer, ModelTensors> modelTensors = new ConcurrentHashMap<>();
         if (mlInput.getInputDataset() instanceof TextDocsInputDataSet) {
             TextDocsInputDataSet textDocsInputDataSet = (TextDocsInputDataSet) mlInput.getInputDataset();
-            int processedDocs = 0;
-            while (processedDocs < textDocsInputDataSet.getDocs().size()) {
+            Tuple<Integer, Integer> calculatedChunkSize = calculateChunkSize(textDocsInputDataSet);
+            CountDownLatch countDownLatch = new CountDownLatch(calculatedChunkSize.v1());
```

Comment:
Can we add detailed comment regarding `calculatedChunkSize.v1()` vs `calculatedChunkSize.v2()`?

Reply:
Can we add comment about this?

Reply:
Added comment here: https://github.com/opensearch-project/ml-commons/pull/1958/files/59f16f0ee6313bc6c12a5630c393fac7ea26cd1e#diff-a5fee3d3840ca1ebf7229073c3a53af51fc3d1a99c960820c93d25613f67532eR76

Reply:
The link seems broken?

Reply:
Yeah, should be caused by new commits, the correct link: https://github.com/opensearch-project/ml-commons/pull/1958/files#diff-a5fee3d3840ca1ebf7229073c3a53af51fc3d1a99c960820c93d25613f67532eR73-R77

Reply:
I was mostly curious to know about v1() vs v2()?

Reply:
v1 and v2 are in a tuple object, and I have stated the return value of v1 and v2 in the java doc of this method, they're chunk size and step size.